test/src/test/java/hudson/slaves/JNLPLauncherTest.java

```diff
@@ -113,6 +118,31 @@ public void testHeadlessLaunch() throws Exception {
         // make sure that onOffline gets called just the right number of times
         assertEquals(1, ComputerListener.all().get(ListenerImpl.class).offlined);
     }
+
+    @Issue("JENKINS-63222")
+    @Test
+    public void testInboundAgentUrlOverride() throws Exception {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        MockAuthorizationStrategy authorizationStrategy = new MockAuthorizationStrategy();
+        authorizationStrategy.grant(Jenkins.ADMINISTER).everywhere().toEveryone();
+        j.jenkins.setAuthorizationStrategy(authorizationStrategy);
+
+        // Override the inbound agent url
+        String inboundAgentUrl = "http://localhost:8080/jenkins";
+        System.setProperty("jenkins.agent.inboundUrl", inboundAgentUrl);
+
+        // Create an agent
+        addTestAgent(false);
+
+        // parse the JNLP page into DOM to inspect the jnlp url argument.
+        JenkinsRule.WebClient agent = j.createWebClient();
+        XmlPage jnlp = (XmlPage) agent.goTo("computer/test/jenkins-agent.jnlp","application/x-java-jnlp-file");
+        Document dom = new DOMReader().read(jnlp.getXmlDocument());
+        Object arg = dom.selectSingleNode("//application-desc/argument[3]/following-sibling::argument[1]");
+        String val = ((org.dom4j.Element)arg).getText();
+        assertEquals(inboundAgentUrl, val);
+        System.clearProperty("jenkins.agent.inboundUrl");
```

Comment:
Or you can use `FlagRule`, assuming the `jenkins-test-harness` version is sufficiently new.

Reply:
I have 2.68 and need 2.69 for `FlagRule`. Is that something I'm allowed to bump in this PR?